File: IIS-core-backend/src/main/java/hr/algebra/iisCoreBackend/dto/ColorDTO.java
package hr.algebra.iisCoreBackend.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties({"internalId"})
@JacksonXmlRootElement(localName = "item")
public class ColorDTO {

    private Long internalId;

    private int id;
    private String name;
    private int year;
    private String color;
    private String pantone_value;
}

File: IIS_client_app/composeApp/src/commonMain/kotlin/hr/algebra/iis_client_app/api/models/ColorDTO.kt
package hr.algebra.iis_client_app.api.models

import kotlinx.serialization.Serializable

@Serializable
data class ColorDTO(
    val internalId: Long? = null,
    val id: Int = 0,
    val name: String = "",
    val year: Int = 0,
    val color: String = "",
    val pantone_value: String = ""
)